server/game/core/gameSteps/AbilityResolver.js

```diff
@@ -88,6 +90,7 @@ class AbilityResolver extends BaseStepWithPipeline {
                 }));
             }
             if (this.context.ability.isActivatedAbility()) {
+                resolveTriggersAfter = [AbilityType.Action, AbilityType.Event, AbilityType.Triggered].includes(this.context.ability.type);
```

Comment:
Same comment as above, I think this is maybe not the right place to do this determination because there may be additional context. For example, I think this logic doesn't work if the ability currently being resolved is a "then" ability, since `context.ability` would be pointing at the original card ability (I _think_).

I'm assuming the goal here is something like "make sure that a card ability gets a triggering window at the end", in which case maybe this should be something that gets controlled by CardAbility.ts? Basically the way it works is that every resolvable ability is a CardAbilityStep. The whole printed ability (potentially multiple steps) is a CardAbility.

So the way it works in practice if there are "then" abilities is that you trigger a CardAbility and it will generate additional CardAbilitySteps to handle the "then" part(s) before it finishes resolving. So it feels like CardAbility is where we need to enforce that trigger resolution happens. Maybe something like a `resolveTriggersAfter = false` constructor parameter on CardAbilityStep, and then CardAbility just overrides it with true when it calls `super()`.